consensus/src/block_storage/block_tree.rs

```diff
@@ -255,6 +258,10 @@ where
         Ok(())
     }
 
+    // @REVIEW: Would it be simpler just to prune all blocks that have round < committed round?
+    // This seems like a lot of fairly complex code to optimize for purning slightly earlier when
+    // there are forks
```

Comment:
I think we were trying to make sure block tree is really a tree, it's one of the few chances we can write BFS lol

Reply:
That seems like it was more important when we used parent relationships to check the commit rule (vs when we embedded that info in the QC)...

Trying to save you from having to write BFS :-)